### PR TITLE
Modified frontend to safely output static images under blog

### PR DESCRIPTION
**Pull request by another-rex**
Fixes image output for the blog. Might still be too strict if we want to allow blog to have images in directories.

Also updates blog to use safe_join to add another layer to preventing escapes.

### CODE REVIEW

File: gcp/appengine/frontend_handlers.py [hunk 1]
```diff
@@ -40,7 +43,8 @@
 _PAGE_SIZE = 16
 _PAGE_LOOKAHEAD = 4
 _REQUESTS_PER_MIN = 30
-_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w-]+$')
+_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w\-]+$')
+_VALID_FILE_NAME = re.compile(r'^[\w\-_]+(\.\w+)*$')
```
Comment (oliverchang) [inline gcp/appengine/frontend_handlers.py:47]: Is _ needed in the first `['\w\-_]` ? `\w` already includes underscores right? 

Also, can you add a comment on what you're trying to ensure here? The regex is starting to get a bit  complicated to understand the intention :)

**Review by oliverchang — APPROVED** [review]
LGTM with some nits

File: gcp/appengine/frontend_handlers.py [hunk 2]
```diff
@@ -40,7 +43,8 @@
 _PAGE_SIZE = 16
 _PAGE_LOOKAHEAD = 4
 _REQUESTS_PER_MIN = 30
-_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w-]+$')
+_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w\-]+$')
```
Comment (andrewpollock) [inline gcp/appengine/frontend_handlers.py:46]: Based on the explanation regex101.com gives me, I'm not seeing the need for escaping the hyphen, are you doing this for a particular reason? Same question for the next line.

**Review by andrewpollock — APPROVED** [review]

File: gcp/appengine/frontend_handlers.py (hunk 2 above)
Comment (andrewpollock) [reply gcp/appengine/frontend_handlers.py:46]: +1 to @oliverchang's comment about comments :-)

File: gcp/appengine/frontend_handlers.py (hunk 1 above)
Comment (another-rex) [reply gcp/appengine/frontend_handlers.py]: Ah you're right, _ is not needed, in fact the regex is not actually needed, but just another layer to stop any invalid names. 

It seems like flask blueprint routes already strip out any / (even url encoded), and we are using `send_from_directory`, so this is actually overkill, so I removed the regex now to simplify things.

File: gcp/appengine/frontend_handlers.py (hunk 2 above)
Comment (another-rex) [reply gcp/appengine/frontend_handlers.py]: You're right it's not needed to be escaped, but if you want to add one more character it becomes a range (e.g. `[A-Z]`), so I just added `\` to be consistent with the other regex. The other one's removed now so I reverted this.

**Pull request merged**